Add FileTree Component

alexcss opened the pull request:
## Add FileTree Component 

* Implemented a new FileTree component

## Description

* Added a file tree component to improve document structure visibility.
* The component displays the documentation hierarchy in a collapsible tree view.
* Includes support for highlighting the current active document.
* Features smooth expand/collapse animations and keyboard navigation.

## Screenshots/GIFs

<img width="1144" alt="image" src="https://github.com/user-attachments/assets/d938f44b-43b9-4fbc-add2-36f24d6d982b" />

## Testing

* Tested component rendering with various document structures
* Ensured proper highlighting of the active document
* Tested expand/collapse functionality
* Verified proper theming and styling

## Checklist

- [x] I have read and understood the contributing guidelines.
- [x] I have followed the style guide and formatting guidelines.
- [x] I have added appropriate comments to explain the changes.
- [x] I have tested my changes thoroughly.

ilanolkies commented:
Go go go!

bguiz commented:
> Go go go!

@ilanolkies coming sooner than you think: https://github.com/rsksmart/rsksmart.github.io/pull/355

ilanolkies commented:
Excellent!!! I send you a dm with the ones I can publish.

alexcss committed:
Add FileTree component for directory visualization

Implemented a new FileTree component to render directory structures and file hierarchies in a collapsible tree format. Included corresponding styles, documentation, and integration into MDX components for usage in markdown files.

alexcss committed:
Enhance FileTree with highlighting, ellipsis, and styling.

Added support for `==` syntax to highlight file or directory names. Introduced ellipsis (`...`) as a placeholder for files and folders. Updated styles and theme variables to support new features.

alexcss committed:
Refactor FileTree icons with react-file-type-icons.

Replaced static icons for files with dynamic icons provided by the `react-file-type-icons` library. Updated styling to integrate the new SVG icons seamlessly. Added `react-file-type-icons` as a new dependency in `package.json`.

alexcss committed:
Merge remote-tracking branch 'origin/main' into feature/file-tree

# Conflicts:
#	package.json
#	src/scss/base/_theme-variables.scss
#	yarn.lock

alexcss committed:
Add `react-file-type-icons@1.0.1` to dependencies in `yarn.lock`

owans reviewed (approved):
LGTM!
